canonical/vanilla-framework
# Example color theme improvements

## Done

More limited version of #5086 that achieves #5086 goals without modifying the back-end or modifying example templates.

Fixes #5069, [WD-11110](https://warthogs.atlassian.net/browse/WD-11110)

## QA

- Open [demo](https://vanilla-framework-5087.demos.haus/docs/examples)
- Open a themed component e.g., [table](https://vanilla-framework-5087.demos.haus/docs/examples/base/table)
    - Verify that the color theme switcher is present and uses Segmented Control component.
    - Verify that the baseline grid toggler works.
    - Inspect the element containing the theme switcher & baseline grid control. Verify that the element (`div.u-example-controls`) is positioned to the bottom right of the page, not its contents. 
    - Switch color themes. Verify that the theme has been changed for the example, segmented control state has been updated to reflect the active theme, and the URL query parameter `theme` has been set to your selected theme.
    - Refresh the browser. Verify that the theme you previously had selected is still active.
    - Small/Medium breakpoints: 
        - Verify that the theme switcher & baseline grid control are neatly positioned in the same element & both visible on screen.
        - Verify that the segmented control segments (color themes) are all on the same horizontal line (do not wrap).
- Open a non-themed component, e.g., [Article Pagination](https://vanilla-framework-5087.demos.haus/docs/examples/patterns/article-pagination).
    - Verify that the color theme switcher is not present & no `theme` query parameter is present in your URL.
    - Verify that the baseline grid switcher is still present & functional.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

![image](https://github.com/canonical/vanilla-framework/assets/46915153/76c4ae3a-1784-414c-8dee-93838b02d829)


[WD-11110]: https://warthogs.atlassian.net/browse/WD-11110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
